osidb/signals.py

```diff
@@ -284,3 +287,44 @@ def update_denormalized_labels_on_affect_change(sender, instance, **kwargs):
             or instance.ps_component != db_instance.ps_component
         ):
             instance.update_denormalized_labels()
+
+
+@receiver(pre_save, sender=Flaw)
+def send_email_on_incident_state_change(
+    sender: type[Flaw], instance: Flaw, **kwargs
+) -> None:
+    """
+    Sends a notification email to key stakeholders whenever the incident
+    state field on Flaw changes.
+    """
+    if (
+        instance._state.adding
+        and instance.major_incident_state == Flaw.FlawMajorIncident.NOVALUE
+    ):
+        # No notification on Flaw creation with empty incident value
+        return
+    elif not instance._state.adding:
+        # No notification on Flaw update with no incident state changes
+        db_instance = Flaw.objects.get(pk=instance.pk)
+        if instance.major_incident_state == db_instance.major_incident_state:
+            return
+
+    flaw_id = instance.cve_id or instance.uuid
+    incident_kind = instance.major_incident_state.split("_")[0].capitalize()
+    flaw_url = reverse("flaw-detail", kwargs={"id": instance.uuid})
+    recipient = EmailSettings().incident_request_recipient
+    payload = {
+        "subject": f"Incident state change for Flaw {flaw_id}",
+        "body": f"The flaw's incident state has been set to {incident_kind}.<br />"
```

Comment:
It seems to me that the message would be confusing in some situations. We take just the first word from the incident state which correctly describes the kind/type of it. However, the last word is important to understand what happened with that incident. The message looks like it was always set which would probably evoked APPROVED but we would get the same message if it was just REQUESTED or if it changed from REQUESTED to REJECTED which is rather that the state was unset. I think that we should reflect the type of change too.